pyccel/ast/builtin_methods/list_methods.py

```diff
@@ -217,34 +208,26 @@ def __init__(self, list_variable, index, new_elem) -> None:
         )
         if not is_homogeneous:
             raise TypeError("Expecting an argument of the same type as the elements of the list")
-        self._index = index
-        self._list_variable = list_variable
-        self._insert_arg = new_elem
-        super().__init__()
-
-    @property
-    def index(self):
-        """
-        Index in which the element will be added.
-
-        Index in which the element will be added.
-        """
-        return self._index
-
-    @property
-    def list_variable(self):
-        """
-        Get the variable representing the list.
+        super().__init__(list_variable, index, new_elem)
 
-        Get the variable representing the list.
-        """
-        return self._list_variable
-
-    @property
-    def insert_argument(self):
-        """
-        Get the argument which is passed to insert().
-
-        Get the argument which is passed to insert().
-        """
-        return self._insert_arg
+#==============================================================================
+class ListExtend(ListMethod):
+    """
+    Represents a call to the .extend() method.
+
+    Represents a call to the .extend() method of an object with a list type,
+    which adds items of an iterable (list, tuple, dictionary, etc) at the end
+    of a list.
+    This method is handled through the call to `_visit_ListExtend` in
+    the semantic stage. It then attempts to construct a `For` loop node with
+    a body that calls `append()`, or a direct `append()` nodes depending on
+    the type of the iterable passed to `extend()`.
+    The extend method is called as follows:
+
+    >>> a = [1, 2, 3]
+    >>> a.extend(range(4, 8))
+    >>> print(a)
+    [1, 2, 3, 4, 5, 6, 7]
+    """
+    __slots__ = ()
```

Comment:
This class inherits its `__init__` method. If you don't define an `__init__` then you will need to document the inherited `__init__` to avoid the error. Alternatively you may want to consider if the inheritance is useful here?

Reply:
Eventually the inheritance is important because of some assertion checks in the superclass `PyccelFunctionDef`
Can I get around this by providing an empty `__init__()` ?

Reply:
Ah that makes sense. You can try. I don't know if pylint will like it though ^^.

Reply:
i guess this update fixes it 7e7177a01d793131338cb53a3b546c77b3a87b35